src/storage/v3/shard_rsm.cpp

```diff
@@ -159,10 +162,15 @@ std::optional<std::map<PropertyId, Value>> CollectAllPropertiesFromAccessor(cons
 }
 
 bool FilterOnVertex(DbAccessor &dba, const storage::v3::VertexAccessor &v_acc, const std::vector<std::string> &filters,
-                    const std::string_view node_name) {
-  return std::ranges::all_of(filters, [&node_name, &dba, &v_acc](const auto &filter_expr) {
-    auto res = ComputeExpression(dba, v_acc, std::nullopt, filter_expr, node_name, "");
-    return res.IsBool() && res.ValueBool();
+                    const std::string_view node_name, const std::optional<EdgeAccessor> &e_acc = std::nullopt) {
+  return std::ranges::all_of(filters, [&node_name, &dba, &v_acc, &e_acc](const auto &filter_expr) {
+    TypedValue result;
+    if (e_acc) {
+      result = ComputeExpression(dba, v_acc, e_acc, filter_expr, "", node_name);
```

Comment:
Don't we want to provide edge_name and node_name here?
Something like:
      result = ComputeExpression(dba, v_acc, e_acc, filter_expr, node_name, edge_name);